File: test/resources/HelloProphecy/prophecy/pipelines/silver_customers_orders/code/silver_customers_orders/graph/CustomerZipCodes.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.utils import *
from prophecy.libs import typed_lit
from silver_customers_orders.config.ConfigStore import *
from silver_customers_orders.udfs.UDFs import *

def CustomerZipCodes(spark: SparkSession, in0: DataFrame, in1: DataFrame, ) -> DataFrame:
    return in0\
        .alias("in0")\
        .join(in1.alias("in1"), (rand_zip_index(col("customer_id")) == col("in1.row_number")), "inner")\
        .select(col("in0.customer_id").alias("customer_id"), col("in0.first_name").alias("first_name"), col("in0.last_name").alias("last_name"), col("in0.phone").alias("phone"), col("in0.email").alias("email"), col("in1.zipcode").alias("zipcode"), col("in0.account_open_date").alias("account_open_date"), col("in0.account_flags").alias("account_flags"))

File: test/resources/HelloProphecy/prophecy_scala/pipelines/silver_customers_orders/code/src/main/scala/io/prophecy/pipelines/silver_customers_orders/graph/CustomerZipCodes.scala
package io.prophecy.pipelines.silver_customers_orders.graph

import io.prophecy.libs._
import io.prophecy.pipelines.silver_customers_orders.udfs.PipelineInitCode._
import io.prophecy.pipelines.silver_customers_orders.udfs.UDFs._
import io.prophecy.pipelines.silver_customers_orders.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object CustomerZipCodes {

  def apply(context: Context, in0: DataFrame, in1: DataFrame): DataFrame =
    in0
      .as("in0")
      .join(in1.as("in1"),
            rand_zip_index(col("customer_id")) === col("in1.row_number"),
            "inner"
      )
      .select(
        col("in0.customer_id").as("customer_id"),
        col("in0.first_name").as("first_name"),
        col("in0.last_name").as("last_name"),
        col("in0.phone").as("phone"),
        col("in0.email").as("email"),
        col("in1.zipcode").as("zipcode"),
        col("in0.account_open_date").as("account_open_date"),
        col("in0.account_flags").as("account_flags")
      )

}
